File: src/main/java/org/zenbot/szolnok/timetable/batch/configuration/TimetableProperties.java
package org.zenbot.szolnok.timetable.batch.configuration;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.zenbot.szolnok.timetable.batch.configuration.properties.TimetableResourceProperties;
import org.zenbot.szolnok.timetable.batch.configuration.properties.TimetableSelectorProperties;

@Data
@ConfigurationProperties("timetable")
public class TimetableProperties {
    private TimetableResourceProperties resource;
    private TimetableSelectorProperties selector;
}

File: szolnok-timetable-batch-utils/src/main/kotlin/org/zenbot/szolnok/timetable/batch/utils/configuration/properties/TimetableProperties.kt
package org.zenbot.szolnok.timetable.batch.utils.configuration.properties

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("timetable")
data class TimetableProperties(
        var resource: TimetableResourceProperties = TimetableResourceProperties(),
        var selector: TimetableSelectorProperties = TimetableSelectorProperties()
)
